### PR TITLE
PP-460/UI-issues

### PR DESCRIPTION
**Pull request by franciscotobar**
## What

- Visual bugs

## Why

- Sometimes, when reloading the dapp, the smartwallets already created are shown twice.

- Currently the balances are not reloaded neither after transactions execution or clicking the refresh button.

## Refs

- [Jira](https://rsklabs.atlassian.net/browse/PP-460)

### CODE REVIEW

**Review by ironFe93 — APPROVED** [review]

File: src/components/ActionBar.tsx [hunk 1]
```diff
@@ -39,30 +41,37 @@ function ActionBar(props: ActionBarProps) {
             );
             console.log(e);
         }
-        setSmartWallets([]);
+        tempSmartWallets.push(...smartWallets);
```
Comment (jurajpiar) [inline src/components/ActionBar.tsx]: I'd prefer immutable approach

File: src/components/ActionBar.tsx [hunk 2]
```diff
@@ -39,30 +41,37 @@ function ActionBar(props: ActionBarProps) {
...
         for (let i = 0; i < tempSmartWallets.length; i += 1) {
             Utils.getSmartWalletBalance(tempSmartWallets[i], state.token!).then(
-                (tempSmartWallet) =>
-                    setSmartWallets((prev) => [...prev, tempSmartWallet])
+                (tempSmartWallet) => {
+                    const exists = smartWallets.find(
+                        (x) => x.address === tempSmartWallet.address
```
Comment (jurajpiar) [inline src/components/ActionBar.tsx]:
```suggestion
                        ({ address }) => address === tempSmartWallet.address
```

File: src/components/ActionBar.tsx [hunk 3]
```diff
@@ -39,30 +41,37 @@ function ActionBar(props: ActionBarProps) {
...
-                        await Utils.tokenBalance(
-                            workerAddress,
-                            state.token!.address
-                        )
-                    )
-                ).toFixed(4);
-                setWorkerBalance(currentWorkerBalance);
             }
         })();
     }, [state.token, updateInfo]);
```
Comment (jurajpiar) [inline src/components/ActionBar.tsx]: this is bad bad... using async functions in `useEffect` is very much discouraged as it results in race conditions. For such situations use `useCallback`.

Comment (franciscotobar) [reply src/components/ActionBar.tsx]: I forgot to remove the async and await. This is not a async function.

File: src/components/ActionBar.tsx (hunk 2 above)
Comment (franciscotobar) [reply src/components/ActionBar.tsx]: updated

File: src/modals/Validate.tsx [hunk 4]
```diff
@@ -109,37 +101,32 @@ function Validate(props: ValidateProps) {
 
     const createSmartWallet = async () => {
         if (state.provider) {
-            dispatch({ type: 'set_loader', loader: true });
+            // dispatch({ type: 'set_loader', show: true });
```
Comment (antomor) [inline src/modals/Validate.tsx]: Is it still necessary? If not I'd remove it.

File: src/modals/Validate.tsx [hunk 5]
```diff
@@ -109,37 +101,32 @@ function Validate(props: ValidateProps) {
...
+            dispatch({
+                type: 'add_smart_wallet',
+                smartWallet: newSmartWallet
+            });
+            dispatch({ type: 'reload', reload: true });
+            if (smartWallet.deployed) {
+                Utils.addLocalSmartWallet(chainId, account, newSmartWallet);
             }
-            dispatch({ type: 'set_loader', loader: false });
+            // dispatch({ type: 'set_loader', show: false });
```
Comment (antomor) [inline src/modals/Validate.tsx]: Is it still necessary? If not I'd remove it.

File: src/modals/Transfer.tsx [hunk 6]
```diff
@@ -123,8 +115,7 @@ function Transfer(props: TransferProps) {
                 tokenAddress: state.token!.address,
                 tokenAmount: Number(fees),
                 transactionDetails: {
-                    retries: 7,
-                    ignoreTransactionReceipt: true
```
Comment (antomor) [inline src/modals/Transfer.tsx]: Why did we decide to wait for the transaction receipt again?

**Review by antomor — COMMENTED** [review]
I tried the dapp and it works like a charm 🪄
Before approving, I've just few doubts

File: src/components/ActionBar.tsx (hunk 1 above)
Comment (franciscotobar) [reply src/components/ActionBar.tsx]: updated

File: src/components/ActionBar.tsx (hunk 3 above)
Comment (franciscotobar) [reply src/components/ActionBar.tsx]: updated

File: src/modals/Validate.tsx (hunk 4 above)
Comment (franciscotobar) [reply src/modals/Validate.tsx]: updated

File: src/modals/Validate.tsx (hunk 5 above)
Comment (franciscotobar) [reply src/modals/Validate.tsx]: updated

File: src/modals/Transfer.tsx (hunk 6 above)
Comment (franciscotobar) [reply src/modals/Transfer.tsx]: updated

**Review by antomor — APPROVED** [review]
LG

**Pull request merged**